### PR TITLE
[alembic] fix FindHDF5.cmake issue for cmake-3.9.0

### PR DESCRIPTION
**Pull request by ousttrue**
cmake versioin 3.9 specific

### CODE REVIEW

File: ports/alembic/portfile.cmake [hunk 1]
```diff
@@ -18,9 +18,15 @@ vcpkg_apply_patches(
...
 
+SET(VCPKG_INSTALLED ${VCPKG_ROOT_DIR}/installed/${TARGET_TRIPLET})
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
     OPTIONS -DUSE_HDF5=ON
+# work around for cmake-3.9 FindHDF5.cmake issue
+# https://gitlab.kitware.com/cmake/cmake/issues/17039
+    -DHDF5_FOUND=1
+    -DHDF5_INCLUDE_DIRS=${VCPKG_INSTALLED}/include
+    -DHDF5_LIBRARIES=${VCPKG_INSTALLED}/lib/hdf5.lib
```
Comment (atkawa7) [inline ports/alembic/portfile.cmake]: @ousttrue Could you use `find_library` and `CURRENT_INSTALLED_DIR`

Comment (ousttrue) [inline ports/alembic/portfile.cmake]: I tried to use find_library in alembic/portfile.cmake,
but ``find_library(HDF5_LIBRARIES NAMES hdf5 PATHS ${CURRENT_INSTALLED_DIR})`` cause error.

```
> .\vcpkg.exe install alembic
The following packages will be built and installed:
    alembic:x86-windows
Building package alembic:x86-windows...
-- CURRENT_INSTALLED_DIR=D:/dev/vcpkg.ousttrue/installed/x86-windows
-- DOWNLOADS=D:/dev/vcpkg.ousttrue/downloads
-- CURRENT_PACKAGES_DIR=D:/dev/vcpkg.ousttrue/packages/alembic_x86-windows
-- CURRENT_BUILDTREES_DIR=D:/dev/vcpkg.ousttrue/buildtrees/alembic
-- CURRENT_PORT_DIR=D:/dev/vcpkg.ousttrue/ports/alembic/.
-- Using cached D:/dev/vcpkg.ousttrue/downloads/alembic-alembic-1.7.1.tar.gz
-- Testing integrity of cached file...
-- Testing integrity of cached file... OK
-- Extracting done
-- Applying patch D:/dev/vcpkg.ousttrue/ports/alembic/fix-hdf5link.patch
-- Applying patch failed. This is expected if this patch was previously applied.
-- Applying patch D:/dev/vcpkg.ousttrue/ports/alembic/fix-hdf5link.patch done
CMake Error: Error required internal CMake variable not set, cmake may not be built correctly.
Missing variable is:
CMAKE_FIND_LIBRARY_PREFIXES
CMake Error: Error required internal CMake variable not set, cmake may not be built correctly.
Missing variable is:
CMAKE_FIND_LIBRARY_SUFFIXES
```

File: ports/alembic/portfile.cmake [hunk 2]
```diff
@@ -21,6 +21,11 @@ vcpkg_apply_patches(
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
     OPTIONS -DUSE_HDF5=ON
+# work around for cmake-3.9 FindHDF5.cmake issue
+# https://gitlab.kitware.com/cmake/cmake/issues/17039
+    -DHDF5_FOUND=1
+    -DHDF5_INCLUDE_DIRS=${CURRENT_INSTALLED_DIR}/include
+    -DHDF5_LIBRARIES=${CURRENT_INSTALLED_DIR}/lib/hdf5.lib
```
Comment (codicodi) [inline ports/alembic/portfile.cmake]: In debug build, `${CURRENT_INSTALLED_DIR}/debug/lib/hdf5_D.lib` should be used.

Comment (ousttrue) [inline ports/alembic/portfile.cmake]: I wrote new patch. It link to debug/lib/hdf5_D.lib in debug build.

**Comment by ras0219-msft** [comment]
Looks good, thanks for the improvement!

**Pull request merged**